packages/datetime/src/common/dateUtils.ts

```diff
@@ -110,6 +110,27 @@ export function isMonthInRange(date: Date, dateRange: DateRange) {
     return start <= day && day <= end;
 }
 
+export function isTimeInRange(date: Date, minDate: Date, maxDate: Date): boolean {
+    const time = getDateOnlyWithTime(date).getTime();
+    const minTime = getDateOnlyWithTime(minDate).getTime();
+    const maxTime = getDateOnlyWithTime(maxDate).getTime();
+
+    const isTimeGreaterThanMinTime = time >= minTime;
+    const isTimeSmallerThanMaxTime = time <= maxTime;
+
+    if (maxTime <= minTime) {
```

Comment:
@cmslewis I'm fine with throwing error solution, I can implement it this way. However, I'm also curious of @giladgray thoughts